fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/runtime/MPJWTAuthExtensionArchivePreparer.java

```diff
@@ -131,6 +134,18 @@ public void process() throws Exception {
         }
 
         addFilterRegistrar();
+
+        if (fraction.getRolesPropertiesMap() != null) {
+            createRolePropertiesFileFromMap();
+        }
+    }
+
+    private void createRolePropertiesFileFromMap() {
+        StringBuilder sb = new StringBuilder();
+        for (Map.Entry<String, String> entry : fraction.getRolesPropertiesMap().entrySet()) {
+            sb.append(entry.getKey()).append('=').append(entry.getValue()).append('\n');
+        }
+        archive.add(new ByteArrayAsset(sb.toString().getBytes()), "WEB-INF/classes/roles.properties");
```

Comment:
Should the filename perhaps be a bit more random, so that we don't risk overwriting an existing one? `roles.properties` is quite predictable. How about `autogenerated-roles.properties`?

Reply:
Sure, I'll have a look